lib/transforms/populate.js

```diff
@@ -36,15 +28,7 @@ module.exports = function (options) {
                         assetGraph.emit('warn', err);
                     }
                     if (resolvedAssetConfig && resolvedAssetConfig.to.isResolved && followRelationsMatcher(resolvedAssetConfig, assetGraph)) {
-                        assetGraph.ensureAssetConfigHasType(resolvedAssetConfig.to, function (err) {
-                            if (err) {
-                                err.message += '\nIncluding asset [' + resolvedAssetConfig.type + ']: ' + resolvedAssetConfig.from.nonInlineAncestor.url;
-                                err.asset = resolvedAssetConfig.to;
-                                err.relation = resolvedAssetConfig;
-                                assetGraph.emit('warn', err);
-                                return cb();
-                            }
-
+                        assetGraph.ensureAssetConfigHasType(resolvedAssetConfig.to).then(function () {
```

Comment:
`ensureAssetConfigHasType` should also resolve with something relevant, like `resolvedAssetConfig`, so we could decouple these functions